scripts/CEE/collect-verbose-ca-logs/script.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# Set variables
+GREEN="\033[32m"
+RESET="\033[0m"
+NAMESPACE="openshift-machine-api"
+CA_NAME="default"
+
+# Get current logVerbosity value
+current_log_verbosity=$(oc get ca $CA_NAME -o jsonpath='{.spec.logVerbosity}')
+
+# Check if current_log_verbosity is empty
+if [ -z "$current_log_verbosity" ]; then
+  echo "Failed to retrieve current logVerbosity value."
+  exit 1
+fi
+
+echo
+echo "CURRENT LOG VERBOSITY = $current_log_verbosity"
+echo
+
+# Update logVerbosity to 6
+oc patch ca $CA_NAME --type='json' -p='[{"op": "replace", "path": "/spec/logVerbosity", "value": 6}]'
+
+updated_log_verbosity=$(oc get ca $CA_NAME -o jsonpath='{.spec.logVerbosity}')
+echo
+echo "UPDATED LOG VERBOSITY = $updated_log_verbosity"
+echo
+
+# Wait for the update to take effect
+echo "Waiting for the log verbosity to update..."
+sleep 30
```

Comment:
In https://github.com/openshift/managed-scripts/pull/187  @devppratik mentioned the idea about instead of blindly sleeping, we might wait for some certain conditions.  Any thought about that?

I haven't tried changing the verbosity by myself, could you help me to understand what it is waiting for? I assume it is waiting for the auto scaler pods to redeploy?

Reply:
It is waiting for the pod to redeploy with the new logVerbosity setting. I am checking if I can add a condition to wait for the new CA POD to be in the READY state.

Reply:
Check condition added in the script.